pkg/mcs/discovery/register.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pingcap/log"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+)
+
+// ServiceRegister is used to register the service to etcd.
+type ServiceRegister struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	cli    *clientv3.Client
+	key    string
+	value  string
+	ttl    int64
+}
+
+// NewServiceRegister creates a new ServiceRegister.
+func NewServiceRegister(ctx context.Context, cli *clientv3.Client, serviceName, serviceAddr string, ttl int64) *ServiceRegister {
+	cctx, cancel := context.WithCancel(ctx)
+	serviceKey := registryPath(serviceName, serviceAddr)
+	return &ServiceRegister{
+		ctx:    cctx,
+		cancel: cancel,
+		cli:    cli,
+		key:    serviceKey,
+		value:  serviceAddr,
```

Comment:
Since we use range scan to discover, can we use "/pd/microserivce/{service name}/registry/{ip:port}" as the key, weight or a json, which contains host properties and starts with weight, as the value?

Reply:
For more details of my comment, please see my comments in the design doc.

Reply:
I'm going to change the value to an interface, which can be implemented by different service itself.

Reply:
sounds good